packages/jaeger-ui/src/actions/deep-dependency-graph.tsx

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import _identity from 'lodash/identity';
+import queryString from 'query-string';
+import { createActions } from 'redux-actions';
+
+import { TDdgActionMeta, TDdgAddStyleAction, TDdgClearStyleAction } from '../model/ddg/types';
+import generateActionTypes from '../utils/generate-action-types';
+
+type TActionTypes = TDdgAddStyleAction | TDdgClearStyleAction;
+
+export const actionTypes = generateActionTypes('@jaeger-ui/DEEP-DEPENDENCY-GRAPH', [
+  'ADD_STYLE_STATE',
+  'CLEAR_STYLE_STATE',
+]);
+
+const addStyleState: (kwarg: TDdgAddStyleAction) => TDdgAddStyleAction = _identity;
+const clearStyleState: (kwarg: TDdgClearStyleAction) => TDdgClearStyleAction = _identity;
+
+function firstParam(arg: string | string[]): string {
+  if (Array.isArray(arg)) {
+    const returnVal = arg[0];
+    console.warn(`Found multiple query parameters: "${arg}", using "${returnVal}"`); // eslint-disable-line no-console
+    return returnVal;
+  }
+  return arg;
+}
+
+export const extractMeta = (): TDdgActionMeta => {
+  const { service, operation, start, end } = queryString.parse(window.location.search);
```

Comment:
Using `window.location.search` instead of the router state is a little strange. Let's discuss ways to avoid this.